site/en/docs/webstore/identify_user/index.md

```diff
@@ -5,84 +5,204 @@ date: 2017-08-30
+
+## Launch the Authorization flow {: #auth-flow}
+
+### Request the "identity" permission {: #identity-permission}
+
+To use the [Chrome Identity API][identity-api], declare the `"identity"` permission in the `manifest.json`.
+
+```json
+{
+  "name": "Google OpenID Connect example",
+  ...
+  "permissions": [
+    "identity"
+  ],
+  ...
+}
+```
+
+### Construct the authorization URL {: #auth-url}
+
+Before the extension can make a request to Google’s OAuth2 endpoint using the Identity API,
+you need to construct an [authorization URL][auth-url]. It must contain several request parameters
+including the client ID and redirect URI. In addition to the `openid` [scope][openid-scopes], you can request
+`profile` information and/or `email`. Update `background.js` to match the following code:
+
+```javascript
+// background.js
+
+let clientId = '<CLIENT_ID>'
+let redirectUri = `https://${chrome.runtime.id}.chromiumapp.org/`
+let nonce = Math.random().toString(36).substring(2, 15)
+
+chrome.action.onClicked.addListener(function () {
+  const authUrl = new URL('https://accounts.google.com/o/oauth2/v2/auth')
+
+  authUrl.searchParams.set('client_id', clientId)
+  authUrl.searchParams.set('response_type', 'id_token')
+  authUrl.searchParams.set('redirect_uri', redirectUri)
+  // Add the OpenID scope. Scopes allow you to access the user’s information.
+  authUrl.searchParams.set('scope', 'openid profile email')
+  authUrl.searchParams.set('nonce', nonce)
+  // Show the consent screen after login
+  authUrl.searchParams.set('prompt', 'consent')
+
+})
+
+```
+
+Replace `<CLIENT_ID>` with the API key generated from the Google console. 
+
+### Retrieve a redirect URL {: #redirect-url}
+
+Now that the extension has the client ID, redirect URI, and OAuth URL, it can initiate Google's
+authentication flow. Call [`identity.launchWebAuthFlow()`][identity-webauthflow] to launch the web auth flow and
+retrieve a redirect URL. 
+
+The redirect URL contains a JSON Web Token (JWT) that identifies the user. To view the requested
+user identity information, you'll need to parse the JWT into a plain JavaScript object. Update
+`background.js` to match the following code:
+
+```javascript
+// background.js
+...
+
+chrome.action.onClicked.addListener(function () {
+
+...
+
+  chrome.identity.launchWebAuthFlow(
+      {
+        url: authUrl.href,
+        interactive: true,
+      },
+      (redirectUrl) => {
+        if (redirectUrl) {
+          // The ID token is in the URL hash
+          const urlHash = redirectUrl.split('#')[1]
+          const params = new URLSearchParams(urlHash)
+          const jwt = params.get('id_token')
+
+          // Parse the JSON Web Token
+          const base64Url = jwt.split('.')[1]
+          const base64 = base64Url.replace('-', '+').replace('_', '/')
+          const token = JSON.parse(atob(base64))
+
+          console.log('token', token)
+        }
+      },
+    )
+})
```

Comment:
```suggestion
// background.js

...

chrome.action.onClicked.addListener(function() {

  ...

  chrome.identity.launchWebAuthFlow(
      {
        url: authUrl.href,
        interactive: true,
      },
      (redirectUrl) => {
        if (redirectUrl) {
          // The ID token is in the URL hash
          const urlHash = redirectUrl.split('#')[1];
          const params = new URLSearchParams(urlHash);
          const jwt = params.get('id_token');

          // Parse the JSON Web Token
          const base64Url = jwt.split('.')[1];
          const base64 = base64Url.replace('-', '+').replace('_', '/');
          const token = JSON.parse(atob(base64));

          console.log('token', token);
        }
      },
    );
});
```

Tweaks to better fit Google's JS style guide. 

* Remove extra newline per [4.1.2 Nonempty blocks: K&R style](https://google.github.io/styleguide/jsguide.html#formatting-nonempty-blocks).
* Add semicolons per [4.3.2 Semicolons are required](https://google.github.io/styleguide/jsguide.html#formatting-semicolons-are-required).
* Remove space between function and parenthese per [4.6.2 Horizontal whitespace](https://google.github.io/styleguide/jsguide.html#formatting-horizontal-whitespace).